docker/lib/dependabot/docker/file_parser.rb

```diff
@@ -167,11 +182,13 @@ def deep_fetch_images_from_hash(json_object)
         images + json_object.values.flat_map { |obj| deep_fetch_images(obj) }
       end
 
+      sig { returns(T::Array[Dependabot::DependencyFile]) }
       def manifest_files
         # Dependencies include both Dockerfiles and yaml, select yaml.
         dependency_files.select { |f| f.type == "file" && f.name.match?(YAML_REGEXP) }
       end
 
+      sig { params(img_hash: T.untyped).returns(T::Array[T.untyped]) }
```

Comment:
Can we be more specific here?
```suggestion
      sig { params(img_hash: T::Hash[String, T.nilable(String)]).returns(T::Array[String]) }
```